src/mongo/MongoScrapbook.ts

```diff
@@ -243,18 +243,24 @@ class FindMongoCommandsVisitor extends MongoVisitor<MongoCommand[]> {
 		if (argumentsContext) {
 			let functionCallContext = argumentsContext.parent;
 			if (functionCallContext && functionCallContext.parent instanceof mongoParser.CommandContext) {
-				const lastCommand = this.commands[this.commands.length - 1];
-				const argAsObject = this.contextToObject(ctx);
-				const argText = EJSON.stringify(argAsObject);
-				lastCommand.arguments.push(argText);
-				let escapeHandled = this.deduplicateEscapesForRegex(argText);
-				let ejsonParsed = {};
 				try {
-					ejsonParsed = EJSON.parse(escapeHandled);
-				} catch (err) { //EJSON parse failed due to a wrong flag, etc.
+					//tslint:disable:no-var-self
+					const me = this;
+					callWithTelemetryAndErrorHandlingSync('mongo.visitArgument', function (this: IActionContext) {
+						this.rethrowError = true;
+						this.suppressErrorDisplay = true;
+						this.suppressTelemetry = true;
+
+						const lastCommand = me.commands[me.commands.length - 1];
+						const argAsObject = me.contextToObject(ctx);
+						const argText = EJSON.stringify(argAsObject);
+						lastCommand.arguments.push(argText);
+						let escapeHandled = me.deduplicateEscapesForRegex(argText);
+						lastCommand.argumentObjects.push(EJSON.parse(escapeHandled));
+					});
+				} catch (err) {
```

Comment:
This seems to use the position from the larger context. 
Have you tested to check if the positions mentioned are correct?

Reply:
It uses the same `ctx` passed in to `contextToObject` a few lines above. Isn't that right?

<img width="540" alt="screen shot 2018-11-21 at 5 06 37 pm" src="https://user-images.githubusercontent.com/11282622/48872709-fdec9480-edaf-11e8-8403-957971cc16af.png">